### PR TITLE
Document minimum IAM policies

### LINKED ISSUE #204: Document minimum IAM requirements
It would be nice to have a documentation listing the minimum IAM permissions to run eksctl.

I'm trying to set this up with a minimum service account, and now I have to add dependencies one by one.

If they were documented as a YAML that would be the best!

### PR DESCRIPTION
**Pull request by martina-if**
Closes #204

@michaelbeaumont @cPu1 I took this from the policies we use for integration tests but we could also use this other format which is more compact: https://github.com/weaveworks/eksctl/issues/204#issuecomment-631630355

- [x] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

File: docs/minimum-iam-policies.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+
+
+EksAllAccess
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
```
Comment (sayboras) [inline docs/minimum-iam-policies.md:112]: This line can be removed I think :)

File: docs/minimum-iam-policies.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+AmazonEC2FullAccess
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": "ec2:*",
+            "Effect": "Allow",
+            "Resource": "*"
```
Comment (sayboras) [inline docs/minimum-iam-policies.md:16]: based on my experience, there might be some question about _exact_ permissions later :). FullAccess on some AWS services (e.g. ec2, lb, cloudwatch, ...) might not be allowed in some orgs such as my company. :)

File: docs/minimum-iam-policies.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            "Action": [
+                "cloudformation:*"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+Ec2AutoScalingLimitedAccess
```
Comment (sayboras) [inline docs/minimum-iam-policies.md:70]: we have _AmazonEC2FullAccess_ above, this is the subset of AmazonEC2FullAccess i think.

**Comment by michaelbeaumont** [comment]
Do we want to link to this doc from the README or on the website?

File: docs/minimum-iam-policies.md (hunk 2 above)
Comment (martina-if) [reply docs/minimum-iam-policies.md:16]: Good point. I don't know how to narrow it down right now so let's leave it like and when someone figures that out we can update it.

File: userdocs/src/usage/minimum-iam-policies.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        {
+            "Effect": "Allow",
+            "Action": "iam:CreateServiceLinkedRole",
+            "Resource": "*",
+            "Condition": {
+                "StringEquals": {
+                    "iam:AWSServiceName": [
+                        "autoscaling.amazonaws.com",
+                        "ec2scheduled.amazonaws.com",
+                        "elasticloadbalancing.amazonaws.com",
```
Comment (kr3cj) [inline userdocs/src/usage/minimum-iam-policies.md:42]: HI @martina-if, I know this is already merged, but we found that this also needs to be included (using eksctl v0.21.0): `"eks.amazonaws.com",`

Comment (martina-if) [reply userdocs/src/usage/minimum-iam-policies.md:42]: Thanks @kr3cj, I updated it [here](https://github.com/weaveworks/eksctl/pull/2267#discussion_r437014109)

If you find more things that need updating feel free to open a PR if you like!

Comment (kr3cj) [reply userdocs/src/usage/minimum-iam-policies.md:42]: Thanks @martina-if , will do.

**Comment by brsolomon-deloitte** [comment]
I don't actually see this page anywhere under the Usage section: https://eksctl.io/usage/

**Comment by brsolomon-deloitte** [comment]
Note to docs readers: this is incorporated at https://eksctl.io/usage/minimum-iam-policies/.

**Pull request merged**